go/engine/login_provisioned_device.go

```diff
@@ -56,119 +59,191 @@ func (e *LoginProvisionedDevice) Run(m libkb.MetaContext) error {
 	}
 
 	m.CDebugf("LoginProvisionedDevice success, sending login notification")
-	m.G().NotifyRouter.HandleLogin(string(m.G().Env.GetUsername()))
+	m.G().NotifyRouter.HandleLogin(e.username.String())
 	m.CDebugf("LoginProvisionedDevice success, calling login hooks")
 	m.G().CallLoginHooks()
 
 	return nil
 }
 
-func (e *LoginProvisionedDevice) run(m libkb.MetaContext) error {
-	// already logged in?
-	in, err := m.G().LoginState().LoggedInProvisioned(m.Ctx())
-	if err == nil && in {
-		if len(e.username) == 0 || m.G().Env.GetUsername() == libkb.NewNormalizedUsername(e.username) {
-			// already logged in, make sure to unlock device keys
-			var partialCopy *libkb.User
-			err = m.G().GetFullSelfer().WithSelf(m.Ctx(), func(user *libkb.User) error {
-
-				// We don't want to hold onto the full cached user during
-				// the whole `unlockDeviceKey` run below, which touches
-				// a lot of (potentially reentrant) code. A partial copy
-				// will suffice. We won't need the non-copied fields
-				// (like the sigchain and ID table).
-				partialCopy = user.PartialCopy()
-				return nil
-			})
-			if err != nil {
-				return err
-			}
-			return e.unlockDeviceKeys(m, partialCopy)
-		}
-	}
+func (e *LoginProvisionedDevice) loadMe(m libkb.MetaContext) (err error) {
+	defer m.CTrace("LoginProvisionedDevice#loadMe", func() error { return err })()
 
 	var config *libkb.UserConfig
-	loadUserArg := libkb.NewLoadUserArg(m.G()).WithPublicKeyOptional().WithForceReload()
 	var nu libkb.NormalizedUsername
+	loadUserArg := libkb.NewLoadUserArgWithMetaContext(m).WithPublicKeyOptional().WithForcePoll(true)
 	if len(e.username) == 0 {
 		m.CDebugf("| using current username")
 		config, err = m.G().Env.GetConfig().GetUserConfig()
-		loadUserArg = loadUserArg.WithSelf(true)
+		if config == nil {
+			m.CDebugf("user config is nil")
+			return errNoConfig
+		}
+		loadUserArg = loadUserArg.WithSelf(true).WithUID(config.GetUID())
```

Comment:
WithSelf() causes loaduser to get self's uid.  WithUID() will be overwritten...

Reply:
My read of loaduser is that the passed in UID overrides the global self UID.